File: Assets/js/LoginPanel.js
#pragma strict

public var usernameField : InputField;
public var passwordField : InputField;
public var feedback : Text;
public var loginButton : LoginButton;
public var onLogin = function() {};

function Start() {
}

function Update() {
}

function HandleRegisterClick() {
	var url : String = String.Format("https://{0}/login/register",ApiManager.GetApiDomain());
	Debug.Log("open url: " + url);
	Application.OpenURL(url);
}

function SetParent(prt : RectTransform) {
	var rt : RectTransform = GetComponent(RectTransform);
	var oldPosition = rt.anchoredPosition;
	var oldScale = rt.localScale;
	
	rt.SetParent(prt);
	
	rt.anchoredPosition = oldPosition;
	rt.localScale = oldScale;	
}

function HandleLogin(resultData : Hashtable) {
	if (!resultData["logged_in"]) {
		feedback.text = TextManager.GetText("Invalid username or password");
		return;
	} else {
		feedback.text = "Success!";
	}
	var userSession : UserSession = UserSession.GetUserSession();
	
	userSession.HandleLogin(resultData);
	
	if (onLogin != null) {
		onLogin();
	}
	
	ClosePanel();
	
}

static function ShowLoginPanel(prefab : LoginPanel, loginButton : LoginButton) : LoginPanel {
	var clone : LoginPanel = Instantiate(prefab, Vector3.zero, Quaternion.identity);
	var canvas : RectTransform = GameObject.Find("Canvas").GetComponent(RectTransform);
	clone.SetParent(canvas);
	clone.loginButton = loginButton;
	return clone;
}

function ClosePanel() {
	if (loginButton != null) {
		loginButton.SyncLoginStatus();
		loginButton.curPanel = null;
	}
	Destroy(this.gameObject);
	
}

function SubmitLogin() {
	var username = usernameField.value;
	var password = passwordField.value;
	var apiManager : ApiManager = ApiManager.GetInstance();
	var arguments : Hashtable = new Hashtable();
	arguments.Add("username",username);
	arguments.Add("password",password);
	
	var options : Hashtable = new Hashtable();
	options.Add("handler",HandleLogin);
	options.Add("method","post");
	options.Add("cacheEnabled",false);
	options.Add("protocol","https");
	
	apiManager.CallApi("login/login",arguments,options);
}

File: Assets/cs/LoginPanel.cs
using UnityEngine;
using System;
using UnityEngine.UI;
using System.Collections;

public class LoginPanel:MonoBehaviour{
	
	public InputField usernameField;
	public InputField passwordField;
	public Text feedback;
	public LoginButton loginButton;
	public Action onLogin = delegate() {};
	
	public void Start() {
	}
	
	public void Update() {
	}
	
	public void HandleRegisterClick() {
		string url = String.Format("https://{0}/login/register",ApiManager.GetApiDomain());
		Debug.Log("open url: " + url);
		Application.OpenURL(url);
	}
	
	public void SetParent(RectTransform prt) {
		RectTransform rt = GetComponent<RectTransform>();
		Vector2 oldPosition = rt.anchoredPosition;
		Vector3 oldScale = rt.localScale;
		
		rt.SetParent(prt);
		
		rt.anchoredPosition = oldPosition;
		rt.localScale = oldScale;	
	}
	
	public void HandleLogin(Hashtable resultData) {
		bool isLoggedIn = (bool) resultData["logged_in"];
		if (!isLoggedIn) {
			feedback.text = TextManager.GetText("Invalid username or password");
			return;
		} else {
			feedback.text = "Success!";
		}
		UserSession userSession = UserSession.GetUserSession();
		
		userSession.HandleLogin(resultData);
		
		if (onLogin != null) {
			Debug.Log ("Call OnLogin");
			onLogin();
		}
		
		ClosePanel();
		
	}
	
	public static LoginPanel ShowLoginPanel(LoginPanel prefab,LoginButton loginButton) {
		LoginPanel clone = (LoginPanel)Instantiate(prefab, Vector3.zero, Quaternion.identity);
		RectTransform canvas = GameObject.Find("Canvas").GetComponent<RectTransform>();
		clone.SetParent(canvas);
		clone.loginButton = loginButton;
		return clone;
	}
	
	public void ClosePanel() {
		if (loginButton != null) {
			loginButton.SyncLoginStatus();
			loginButton.curPanel = null;
		}
		Destroy(this.gameObject);
		
	}
	
	public void SubmitLogin() {
		string username = usernameField.text;
		string password = passwordField.text;
		Action<Hashtable> handleLogin = HandleLogin;
		ApiManager apiManager = ApiManager.GetInstance();
		Hashtable arguments = new Hashtable();
		arguments.Add("username",username);
		arguments.Add("password",password);
		
		Hashtable options = new Hashtable();
		options.Add("handler", handleLogin);
		options.Add("method", "post");
		options.Add("cacheEnabled", false);
		options.Add("protocol", "https");
		
		StartCoroutine(apiManager.CallApi("login/login",arguments,options));
	}


}
